sdc_client/client.py

```diff
@@ -338,6 +338,135 @@ async def execute_requests(queries_: List[Tuple[IStreamSets, str]]):
+              for pipeline in pipelines],
+            return_exceptions=return_exceptions
+        )
+
+
+async def update_pipeline_async(pipelines: List[IPipeline], configs: List[Dict], return_exceptions=False):
+    clients = [_get_async_client(ss) for ss in {p.get_streamsets() for p in pipelines}]
+    async with _AsyncClientsManager(clients) as manager:
+        return await asyncio.gather(
+            *[asyncio.create_task(manager.clients[pipeline.get_streamsets().get_id()].update_pipeline(pipeline.get_id(), config))
+              for pipeline, config in zip(pipelines, configs)],
+            return_exceptions=return_exceptions
+        )
+
+
+def _update_pipelines_async(pipelines: List[IPipeline], set_offset: bool = False):
+    if set_offset:
+        asyncio.run(set_offsets_async(pipelines))
+    new_configs = [pipeline.get_streamsets_config() for pipeline in pipelines]
+    old_configs = asyncio.run(get_pipelines_async(pipelines))
+    for config in new_configs:
+        new_uuid = [old_config['uuid'] for old_config in old_configs if config['title'] == old_config['title']][0]
+        config['uuid'] = new_uuid
+    asyncio.run(update_pipeline_async(pipelines, new_configs))
+
+
+def _stop_running(pipelines: List[IPipeline]) -> List[IPipeline]:
+    pipelines_info = asyncio.run(get_pipeline_statuses_async(pipelines))
+    pipeline_statuses = {pipeline.get_id(): pipelines_info[0][pipeline.get_id()]['status'] for pipeline in pipelines}
+    pipelines_running = [p for p in pipelines if pipeline_statuses[p.get_id()] in
+                         (IPipeline.STATUS_RUNNING, IPipeline.STATUS_STARTING, IPipeline.STATUS_RETRY)]
+    try:
+        asyncio.run(stop_async(pipelines_running))
+        time.sleep(2)
+    except Exception as e:
+        inject.instance(ILogger).debug(f'Could not stop pipelines {[p.get_id() for p in pipelines]} asynchronously: {str(e)}')
+        for pipeline in pipelines:
+            force_stop(pipeline)
+    return pipelines_running
+
+
+def update_async(pipelines: List[IPipeline]):
+    try:
+        pipelines_running = _stop_running(pipelines)
+        _update_pipelines_async(pipelines)
+        asyncio.run(start_async(pipelines_running))
+    except Exception as e:
+        raise StreamsetsException(f'Pipelines could not be updated due to the errors: {str(e)}') from e
+
+
+def move_to_streamsets_async(rebalance_map: Dict[IPipeline, IStreamSets]):
+    pipelines = list(rebalance_map)
+    try:
+        pipelines_running = _stop_running(pipelines)
+        asyncio.run(delete_async(pipelines))
+        for pipeline_, streamsets_ in rebalance_map.items():
+            pipeline_.set_streamsets(streamsets_)
+        asyncio.run(create_async(pipelines))
+        _update_pipelines_async(pipelines, set_offset=True)
+        asyncio.run(start_async(pipelines_running))
+    except Exception as e:
+        raise StreamsetsException(f'StreamSets could not be balanced due to the errors: {str(e)}') from e
```

Comment:
This method looks specific for StreamsetsBalancer, maybe is better move it inside StreamsetsBalancer, or it can be used in other cases?

Reply:
Yes, it could be there, I guess. Just don't want to have these `asyncio.run(...)` calls in the balancer's code